# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382230065 25200
#      Sat Oct 19 17:47:45 2013 -0700
# Node ID 4d8ddc44d1f26ca50a5efc23ea165222d59a9b80
# Parent  6516fc672fbb07a6866373ca2d0e7d0940791d02
colorbar.m: Avoid segfault in colorbar (bug #40308)

* scripts/plot/draw/colorbar.m: Use low-level form of image
command so that newplot is not called and axes are not switched
from true plotting axes to colorbar axes object.

diff --git a/scripts/plot/draw/colorbar.m b/scripts/plot/draw/colorbar.m
--- a/scripts/plot/draw/colorbar.m
+++ b/scripts/plot/draw/colorbar.m
@@ -196,28 +196,31 @@ function h = colorbar (varargin)
                              "activepositionproperty", "position",
                              "position", cpos);
     addproperty ("location", cax, "radio",
                  "eastoutside|east|westoutside|west|northoutside|north|southoutside|south",
                  loc);
     addproperty ("axes", cax, "handle", ax);
 
     if (vertical)
-      hi = image (cax, [0,1], [cmin, cmax], [1 : clen]');
+      ## Use low-level form to avoid calling newplot which changes axes
+      hi = image (cax, "xdata", [0,1], "ydata", [cmin, cmax],
+                       "cdata", [1 : clen]');
       if (mirror)
         set (cax, "xtick", [], "xdir", "normal", "ydir", "normal",
                   "ylim", cext, "ylimmode", "manual",
                   "yaxislocation", "right", args{:});
       else
         set (cax, "xtick", [], "xdir", "normal", "ydir", "normal",
                   "ylim", cext, "ylimmode", "manual",
                   "yaxislocation", "left", args{:});
       endif
     else
-      hi = image (cax, [cmin, cmax], [0,1], [1 : clen]);
+      hi = image (cax, "xdata", [cmin, cmax], "ydata", [0,1],
+                       "cdata", [1 : clen]);
       if (mirror)
         set (cax, "ytick", [], "xdir", "normal", "ydir", "normal",
                   "xlim", cext, "xlimmode", "manual",
                   "xaxislocation", "top", args{:});
       else
         set (cax, "ytick", [], "xdir", "normal", "ydir", "normal",
                   "xlim", cext, "xlimmode", "manual",
                   "xaxislocation", "bottom", args{:});
